example/battery.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:bluez/bluez.dart';
+
+void main() async {
+  var client = BlueZClient();
+  await client.connect();
+
+  var adapter = client.adapters.first;
+  var bpm = adapter.batteryProviderManager;
+
+  var provider = await bpm.registerBatteryProvider();
+  var battery =
+      await provider.addBattery(percentage: 80, source: 'Dummy Battery');
+
+  battery.percentage = 10;
+
+  while (true) {}
```

Comment:
This line shouldn't be required - the Dart process stays open until the BlueZ client is closed, so in the example it never occurs.

Reply:
The while loop line or the battery percentage line?

Reply:
The while loop.

Reply:
Having the loop like this means it sits there using a lot of CPU, without it the process is just blocking for socket input and closes when the last socket closes.

Reply:
Gotcha, GitHub didn't quite make things clear. I've removed the line.